### ISSUE TITLE
race condition on s3 bucket create

### ISSUE THREAD

**Issue by xchapter7x** [terraform-providers/terraform-provider-aws#877]
### Terraform Version
terraform_0.9.8_linux_amd64.zip

### Affected Resource(s)
Please list the resources as a list, for example:
- resource_aws_s3_bucket

### Terraform Configuration Files
```hcl
resource "aws_s3_bucket" "pcf-bosh" {
    bucket = "${var.prefix}-pcf-bosh"
    acl = "private"
    force_destroy= true

    tags {
        Name = "${var.prefix}-pcf-bosh"
        Environment = "${var.prefix}"
    }
}

resource "aws_s3_bucket" "pcf-buildpacks" {
    bucket = "${var.prefix}-pcf-buildpacks"
    acl = "private"
    force_destroy= true

    tags {
        Name = "${var.prefix}-pcf-buildpacks"
        Environment = "${var.prefix}"
    }
}
```

### Debug Output
https://gist.github.com/xchapter7x/33f777f55f3033bb0ed7c239c70fa6a6

### Expected Behavior
Given a call to create a new bucket during terraform apply
When the buckets are successfully created
Then terraform should exit successfully without error

### Actual Behavior
Given a call to create a new bucket during terraform apply
When the buckets are successfully created
Then terraform intermittently exits with `NoSuchBucket: The specified bucket does not exist` error

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply` * 10 and you will see this error on a handful of attempts

### Important Factoids
It feels like the follow on calls to the bucket, which happen after create bucket (https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_s3_bucket.go#L537)
should be wrapped with calls to `WaitUntilBucketExists` provided by the aws sdk.
This should avoid any potential races.

### References
The issue I am experiencing was referenced as a side note in this issue (https://github.com/terraform-providers/terraform-provider-aws/issues/291), but I do not see a issue created specifically for this.

**Comment by xchapter7x** [comment]
began this WIP PR... just investigating how you do your acceptance tests so i can add some before removing WIP marker... 
Let me know your thought sin the mean time

https://github.com/terraform-providers/terraform-provider-aws/pull/891

**Comment by radeksimko** [comment]
Addressed in #891
